### DISCUSSION TITLE
Get-Deployoption

### DISCUSSION THREAD

**Discussion by jeroenlambrichts** [vmware/govmomi#2933] [Community Help]
Dear All, 

We start with: 

$cred = Get-Credential
Connect-VIServer -Server server.domain.corp -Credential $cred -AllLinked

When we run command Get-DeployOption, we receive a list:

PS C:\Windows\system32> Get-DeployOption

Key           Value
---           -----
ssl-protocols -TLSV1,+TLSV1.2,-SSLV3,-TLSV1.1
vlan-id       xxxx
vlan-id       xxxx
ssl-protocols -TLSV1,-SSLV3,+TLSV1.2,-TLSV1.1
vlan-id       xxxx
ssl-protocols -TLSV1,+TLSV1.2,-TLSV1.1,-SSLV3
vlan-id       xxxx
ssl-protocols +TLSV1.2,-TLSV1.1,-TLSV1,-SSLV3

But when we run it a second time, it throws an error: 

Get-DeployOption : 23/08/2022 14:32:40  Get-DeployOption                The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.
At line:1 char:1
+ Get-DeployOption
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-DeployOption], VimException
    + FullyQualifiedErrorId : Core_BaseCmdlet_UnknownError,VMware.DeployAutomation.Commands.GetDeployOption

Get-DeployOption : 23/08/2022 14:32:40  Get-DeployOption                The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.
At line:1 char:1
+ Get-DeployOption
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-DeployOption], VimException
    + FullyQualifiedErrorId : Core_BaseCmdlet_UnknownError,VMware.DeployAutomation.Commands.GetDeployOption

Get-DeployOption : 23/08/2022 14:32:40  Get-DeployOption                The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.
At line:1 char:1
+ Get-DeployOption
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-DeployOption], VimException
    + FullyQualifiedErrorId : Core_BaseCmdlet_UnknownError,VMware.DeployAutomation.Commands.GetDeployOption

Key           Value
---           -----
vlan-id       xxxx
ssl-protocols +TLSV1.2,-TLSV1.1,-TLSV1,-SSLV3

It seems that it disconnects from all vCenters after running the initial command. Is this possible or by design? 

I also unistalled the VMware.PowerCLI module from the entire system and cleaned everything under "C:\Program Files\WindowsPowerShell\" whereafter I performed a new installation with the latest update(s)

PS C:\Windows\system32> Get-PowerCLIVersion

PowerCLI Version
----------------
   VMware.PowerCLI 12.7.0 build 20091289
---------------
Component Versions
---------------
   VMware Common PowerCLI Component 12.7 build 20067789
   VMware Cis Core PowerCLI Component PowerCLI Component 12.6 build 19601368
   VMware VimAutomation VICore Commands PowerCLI Component PowerCLI Component 12.7 build 20091293
   VMWare ImageBuilder PowerCLI Component 7.0 build 19599828
   VMWare AutoDeploy PowerCLI Component 7.0 build 19599828

PowerCLI Version
----------------
   VMware.PowerCLI 12.7.0 build 20091289
---------------
Component Versions
---------------
   VMware Common PowerCLI Component 12.7 build 20067789
   VMware Cis Core PowerCLI Component PowerCLI Component 12.6 build 19601368
   VMware VimAutomation VICore Commands PowerCLI Component PowerCLI Component 12.7 build 20091293
   VMWare ImageBuilder PowerCLI Component 7.0 build 19599828
   VMWare AutoDeploy PowerCLI Component 7.0 build 19599828

PowerCLI Version
----------------
   VMware.PowerCLI 12.7.0 build 20091289
---------------
Component Versions
---------------
   VMware Common PowerCLI Component 12.7 build 20067789
   VMware Cis Core PowerCLI Component PowerCLI Component 12.6 build 19601368
   VMware VimAutomation VICore Commands PowerCLI Component PowerCLI Component 12.7 build 20091293
   VMWare ImageBuilder PowerCLI Component 7.0 build 19599828
   VMWare AutoDeploy PowerCLI Component 7.0 build 19599828

PowerCLI Version
----------------
   VMware.PowerCLI 12.7.0 build 20091289
---------------
Component Versions
---------------
   VMware Common PowerCLI Component 12.7 build 20067789
   VMware Cis Core PowerCLI Component PowerCLI Component 12.6 build 19601368
   VMware VimAutomation VICore Commands PowerCLI Component PowerCLI Component 12.7 build 20091293
   VMWare ImageBuilder PowerCLI Component 7.0 build 19599828
   VMWare AutoDeploy PowerCLI Component 7.0 build 19599828
   
   Previous version was at 12.3

Any help would be greatly appreciated. 

Thanks in advance. 

Best regards,
Jeroen Lambrichts

**Comment by tenthirtyam** [comment]
This would best be served on the [VMware PowerCLI Discussions](https://communities.vmware.com/t5/VMware-PowerCLI-Discussions/bd-p/2805) within the VMware Communities as this is Discussions area is for `vmware/govmomi`.

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by jlambrichts** [reply]
Hi Ryan, 

I will post it on there. 

Thanks. 

Jeroen
